events/events_test.go

```diff
@@ -23,51 +28,86 @@ type TestSubscriber struct {
 }
 
 func NewTestSubscriber() *TestSubscriber {
-	my := &TestSubscriber{
+	sub := &TestSubscriber{
 		lock:    &sync.RWMutex{},
 		results: []Event{},
 	}
-	my.Rx = make(chan Event, 100)
-	return my
+	sub.Rx = make(chan Event, 100)
+	return sub
 }
 
-func (ts *TestSubscriber) Run(bus *EventBus) {
+func (ts *TestSubscriber) Run(ctx context.Context, bus *EventBus) {
```

Comment:
Any reason not to make `ctx` a pointer?

Reply:
We're not using a pointer because the value never changes. We always want to share a static `context` across the entire application.

It's also worth noting the following documentation from the standard library `context` package...

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each 
> function that needs it. The Context should be the first parameter, typically named `ctx`.

So, while we can technically store this `context.Context` value within a struct (and I've seen plenty of [folks do this](https://github.com/joyent/pg_prefaulter/blob/master/agent/agent.go#L46)), it's not recommended. I can't give you a good reason yet why it's not. I'm on the fence, it always looks handy to store the `context.Context` on a struct.